src/nix-env/nix-env.cc

```diff
@@ -501,9 +501,17 @@ static bool keep(PackageInfo & drv)
     return drv.queryMetaBool("keep", false);
 }
 
+static void setMetaFlag(EvalState & state, PackageInfo & drv,
```

Comment:
This is unchanged, simply relocated earlier in the file so it can be used in `installDerivations`